blockchain/chain_sync/src/peer_manager.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use libp2p::core::PeerId;
+use log::debug;
+use parking_lot::RwLock;
+use std::collections::HashSet;
+
+/// Thread safe peer manager
+#[derive(Default)]
+pub struct PeerManager {
+    /// Hash set of full peers available
+    full_peers: RwLock<HashSet<PeerId>>,
+}
+
+impl PeerManager {
+    /// Adds a PeerId to the set of managed peers
+    pub fn add_peer(&self, peer_id: PeerId) {
+        debug!("Added PeerId to full peers list: {}", &peer_id);
+        self.full_peers.write().insert(peer_id);
+    }
+
+    /// Returns true if peer set is empty
+    pub fn is_empty(&self) -> bool {
+        self.full_peers.read().is_empty()
```

Comment:
Using sth like parking_lot here will make the executor block, if this is called from an `async` function. I suggest using `async_std::sync::RwLock` instead, which avoids this, as it integrates with the executor.

Reply:
hmm interesting, will fix now. thanks :)

Reply:
https://github.com/ChainSafe/forest/pull/252/commits/43f56ab7a90da01cff787853a2ff9893bf5f0ac1 sufficient? Thanks so much for this review by the way !

Reply:
yes, looks good.